### PR TITLE
refactor: Use prost's enum helper functions for converting from ints

### PR DESCRIPTION
**Pull request by carols10cents**
Takes advantage of built-in prost functionality [that @tustvold pointed out](https://github.com/influxdata/influxdb_iox/pull/968#discussion_r593145883)!

### CODE REVIEW

File: tests/end_to_end_cases/storage_api.rs [hunk 1]
```diff
@@ -277,7 +278,7 @@ async fn measurement_fields_endpoint(
 
     let field = &fields[0];
     assert_eq!(field.key, "value");
-    assert_eq!(field.r#type, DataType::Float as i32);
+    assert_eq!(field.r#type(), FieldType::Float);
```
Comment (carols10cents) [inline tests/end_to_end_cases/storage_api.rs:281]: I think this spot was actually a type error!

**Review by pauldix — APPROVED** [review]
nice simplification

File: tests/end_to_end_cases/storage_api.rs (hunk 1 above)
Comment (alamb) [reply tests/end_to_end_cases/storage_api.rs:281]: 👍

**Review by alamb — APPROVED** [review]
Thank you @carols10cents  this is *muuuuch* better. Given I was the one who wrote the original janky code (new favorite word) when I didn't know any better I am going to presumptively merge this  PR.

**Pull request merged**